pgvectorscale/src/access_method/graph.rs

```diff
@@ -419,6 +437,17 @@ impl<'a> Graph<'a> {
                         continue;
                     }
 
+                    // Does it contain essential labels?
+                    if let Some(labels) = labels {
```

Comment:
performance: we should compute the intersection between existing neighbor's labels and `labels` before the for loop on line 434. Then we can use that as the `c` instead of existing_neighbors.get_labels(). Probably worth it to get a shorter list. Can also leave a code comment and address later.